<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Joelle Barido-Sottani, Rachel C. M. Warnock" />

<meta name="date" content="2022-07-19" />

<title>The FossilSim package</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">The FossilSim package</h1>
<h4 class="author">Joelle Barido-Sottani, Rachel C. M. Warnock</h4>
<h4 class="date">2022-07-19</h4>



<p>This vignette provides a <a href="#quick-start">Quick start</a> and a
brief introduction to the package input and output.</p>
<div id="contents" class="section level2">
<h2>Contents</h2>
<ul>
<li><a href="#background-and-motivation">Background and
motivation</a></li>
<li><a href="#installing-the-package">Installing the package</a></li>
<li><a href="#quick-start">Quick start</a></li>
<li><a href="#fossilsim-input">FossilSim input</a></li>
<li><a href="#fossilsim-output">FossilSim output</a></li>
<li><a href="#what-next">What next?</a></li>
</ul>
</div>
<div id="background-and-motivation" class="section level2">
<h2>Background and motivation</h2>
<p><code>FossilSim</code> is an R package for simulating species
taxonomy and fossil occurrence data in a phylogenetic framework.
Simulations can be used to address many questions in phylogenetics and
palaeobiology, and are especially useful for assessing the performance
of different methods, since the true underlying parameters
(e.g. diversification and fossil recovery rates) are known. Tara Smiley
provides an excellent demonstration of the value of simulations in
palaeobiology in this <a href="https://www.cambridge.org/core/blog/2018/03/19/testing-the-fossil-record-how-simulations-help-us-understand-the-relative-roles-of-diversification-and-preservation-underlying-diversity-gradients-in-deep-time/">blog
post</a>. <code>FossilSim</code> output can be easily parsed for
downstream analysis using R, or other software packages including <a href="http://www.beast2.org/">BEAST2</a> and <a href="https://revbayes.github.io">RevBayes</a>. The package provides a
wide range of flexible models, including models of interval-, lineage-
and environment-dependent fossil recovery, in addition to plotting
functions that can be used to visualise the output and produce
publication quality figures.</p>
</div>
<div id="installing-the-package" class="section level2">
<h2>Installing the package</h2>
<p><code>FossilSim</code> is available on CRAN or the latest development
version can be installed from <a href="https://github.com/">GitHub</a>
using the package <code>devtools</code>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># to install the package via CRAN</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;FossilSim&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># to install the package via GitHub</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;rachelwarnock/fossilsim&quot;</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># load the package into the current working environment</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(FossilSim)</span></code></pre></div>
<div id="package-dependencies" class="section level3">
<h3>Package dependencies</h3>
<p>The installation above will automatically install the package
dependencies, <code>ape</code> and <code>TreeSim</code>.</p>
</div>
<div id="calling-functions-from-fossilsim-and-other-packages" class="section level3">
<h3>Calling functions from FossilSim and other packages</h3>
<p>Once the package is loaded into your current working environment you
can call the package functions directly,
e.g. <code>sim.fossils.poisson()</code>. It is also possible to call
functions in R without loading the package into your working
environment, e.g. <code>FossilSim::sim.fossils.poisson()</code>.
Throughout this vignette and other documentation associated with
<code>FossilSim</code>, we call the <code>FossilSim</code> functions
from the current working environment but use the <code>::</code> format
to call functions from other packages. This is done to make the source
of all functions as clear as possible.</p>
</div>
<div id="quick-start" class="section level3">
<h3>Quick start</h3>
<p>Simulating data using <code>FossilSim</code> can be as simple as the
following code snippets.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># simulate a tree using ape</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>tips <span class="ot">=</span> <span class="dv">8</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>t <span class="ot">=</span> ape<span class="sc">::</span><span class="fu">rtree</span>(tips)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># simulate fossils using fossilsim</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>rate <span class="ot">=</span> <span class="dv">2</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>f <span class="ot">=</span> <span class="fu">sim.fossils.poisson</span>(<span class="at">rate =</span> rate, <span class="at">tree =</span> t)  </span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co"># plot the output</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(f, <span class="at">tree =</span> t)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAAATlBMVEUAAAC/v7+/v8q/yuC/1erKv7/Kv8rKv9XKyuDK4PTVv8rVv9XV1crV6v/gyr/gysrg9P/q1b/q///04Mr0////6tX/9OD//+r///T////qTH6cAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAIRUlEQVR4nO3d21IbRxRGYUicA4ljmxCwef8XDYNwldG41K1Wd8/8S+u7SOyLjHbvZR0wErl5FtrN1gNoLAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMxw58893Wg2yHfXT26aqwV8A+XRX2Ctinq8JeAft0VdgrYJ+uCnsF7NNVYa+Afboq7BWwT1eFvQL26aqwV8A+XRX2Ctinq8JeAft0VdgrYJ+uCnsF7NNVYa+Afboq7BWwT1eFvQL26aqwV8A+XRX2Ctinq8JeAft0VVAruFnZeqLtoVawOgzqdG1QKzDwGmoFBl5DrcDAa6gVGHgNtQIDr6FWYOA11AoMvIZagYHXUCsw8BpqBQZeQ63AwGuoFRh4DbUCA6+hVmDgNdQK5gTOWlnWtAVTAoe9jSBq2JIZgdPeKJI0a9GEwHFvBQoatWwdeKAtDtggZc4qEw4TFXeRNGvRjMOE9TXw+TeStbKsaQtQh+kEtRPUYTpB7QR1mE5QO0EdphPUTlCH6QS1E9RhOgnZCetvl2YK2UndmCGHmSpkJwZuFbITA7cK2YmBW4XsxMCtQnZi4FYhOzFwq5CdGLhVyE4M3CpkJwZuFbITA7cK2YmBW4XsxMCtQnZi4FYhOzFwq5CdGLhVyE4M3CpkJwZuFbITA7cK2YmBW4XsxMCtQnZi4FYhOzFwq5CdGLhVyE4M3CpkJwZuFbITA7cK2YmBW4XsxMCtQnZSGdhPIK6EnDdkzB0K2VzImDsUsrmQMXcoZHMhY+7Q0M21/IxPXxr1NTbwDq90bQwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGCw3sB4ZrZQb2I+HVIgP7of96KYH9cQ+NQgIfXdeO1SID+xxcLzOwr6KrhQZWLQPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwcwP7s62mmxx45K3pZwwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGGx34/Sd/DTzd4MBHn+028HRjAx9/et/A080I7E9q2NDYlXsP3tzglfscvLXRK/dV9MbmrtzA02UH9mVbUXjgztcDMjCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzBcUuCjD7K9/NbARUGBjz5q7AfKq+QEfl/UnxhQaXLg7qaOnyhnQ++L2rdS0IreFzVwnaQVHf28D19F14ha0c3x76Km30b2irKnnyJ7RdnTT5G9otnTn3xZ1+El39v1e754NPA5N3fqhXuHF/Vv1+/65YGBz7i1U1+adfiybcjf4Bi47oY20mHyyy+xoXmBX/9xaukdivRue7hop+tsY27g8nPwhfMcLtG1r4HPuqGTi+9Q5XCJnn0NfN4NnbzBbs+Yfpn03fTAeXInXxi4KHfyhYGLcidfGLgod/KFgYtyJ18YuCh38oWBi3InXxi4KHfyhYGLcidfGLgod/KFgYtyJ18YuCh38kXF9K1/cX9478wZN7RTuZMvytO3fuvt6NvuuWvKnXxRnL71m+fHb6zIXVPu5Is+b3yqsfVJm+VOXqM5z3Ha3DXlTl6l+e53dNfNXVPu5HWaH17f3/dz15Q7eaXmp0+/TLomuWvKnXyq3DXlTj5V7ppyJ58qd025k0+Vu6bcyafKXVPu5FPlril38qly15Q7+VS5a8qdfKrcNeVOPlXumnInnyp3TbmTT1Xx1pHWKw9+N4GBq5TWdOF3JZv+08obGHhtkMKaLn5fQdNQdbcw7tIknd/h9TOjJh903avS3mh0XgP30R7Jh+gM7ZEG9zVwJ+2RBv+PCQwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMw3NjAt7rM5QlGBt56OwCXRxgbeODFr4GB4QwMZ2A4A8MZGM7AcAaGMzCcgeH2Hlg7YGA4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAzXP/C3T7e3H95+/XD3/Hx/e/tP9xu5Co89Ntc/8MOHl8aHwZ5+u3t+/PW/b5+/dL+VK/D1ry9Pf/x76VXGPEQv99yX+/Lnj3evvzz8Vud5+rPHXWNI4K9/v/7Be7h7ONyDPxm4wePLE939xY/RIwK/PLYs/3r5E3h4Dv7lo4Eb7Dbw0++Hx5WH5RPMr2kvH/Ma7fUh+scXBstD9IfnDi8VrtFeX2Tdv95xD0/Dfpl0iZ1+maRdMTCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdguP8BeKAEojPDyNoAAAAASUVORK5CYII=" /><!-- --></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># simulate taxonomy using fossilsim</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>beta <span class="ot">=</span> <span class="fl">0.5</span> <span class="co"># probability of symmetric speciation</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>lambda.a <span class="ot">=</span> <span class="fl">0.1</span>  <span class="co"># rate of anagenesis</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>s <span class="ot">=</span> <span class="fu">sim.taxonomy</span>(<span class="at">tree =</span> t, <span class="at">beta =</span> beta, <span class="at">lambda.a =</span> lambda.a)  </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># plot the output</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(s, <span class="at">tree =</span> t, <span class="at">legend.position =</span> <span class="st">&quot;bottomright&quot;</span>)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAAAmVBMVEUAAAAAADoAAGYAAIAAOpAAQIAAZrYAgAAAgEAAgIA6AAA6ADo6kNtAgABmAABmADpmZjpmtrZmtv9/f/9/v/9//39//79///+AAACAAECAAICAQACAgACQOgCQkGaQ2/+2ZgC2tma225C2/7a2//+//3/bkDrb/7bb////f3//f7//f///tmb/v3//25D//3///7b//9v///9IluA8AAAACXBIWXMAAA7DAAAOwwHHb6hkAAALVElEQVR4nO3cDZ/iRgGAcbqeq17Xq8V37fZorXgqbnf5/h9O8kpCwjIMScg8ef4/bSmbKzPzNLAhgdVeaKt7D0DjMjCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAyHDbxqufdo7gc79brtnw6ws7wMO3UDF7BTN3ABO/U68K8PsLO8DDt19+ACduoGLmCnbuACduoeBxfYU2fPLgh7CdizC8JeAvbsgrCXgD27IOwlYM8uCHsJ2LMLwl4C9uyCsJeAPbsg7CVgzy4IewnYswvCXgL27IJQlqD/XWfK7G5AWYJVv3sP6/4oS1An/bmJMrsbUJbAwGdQlsDAZ1CWoA78XRNldjegLIF78BmUJTDwGZQlMPAZqCXwOLgLtQSdyaBmFwe1BAbuQi2BgbtQS2DgLtQSGLgLtQQG7kItgYG7UEtg4C7UEhi4C7UEBu5CLYGBu1BL0A08ontMMEIq4wzSDfzzaFJZuFTGGcTAXamMM4iBu1IZZ5Bu4O9Gk8rCpTLOIO7BXamMM4iBu1IZZxADd6UyziAeB3elMs4gqMkMJI01ge1VU0phTcp6/7kohclMLYU1MfANEliT+gnYwBESWJM68F8vSWAyk0tgTdyDb5HCmvgafIMU1sTAN0hjTTwOjpbImoQNM5HJTCqRNTFwrETWxMCxElkTA8dKZE0MHCuRNTFwrETWxMCxElkTA8dKZE0MHCuRNTFwrETWxMCxElkTA8dKZE0MHCuRNTFwrETWxMCxElkTA8dKZE0MHCuRNTFwrETWxMCxElkTA8dKZE0MHCuRNQkM7JWXHYnMNzDw5UunF3ftdCLzNXCsROZr4FhjzXfgl7zAwBc/f7i8TyCOHnggYQ/qHtwxeuA/D8LAsQwMxwo82JM9x+iBfzOIpWUZDmoPVpeB4QwMN+rKXXeo669GYxg38Az/TUtjYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYLg0Avsxl2gpBC67/SGQgZsMDJdA4Pqp18ARUgr8yzAGbkopsHtwhAQC+xp8CwPDpRDY4+AbpBFY0QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGmzSwn+md3oSBy5T/7GPgsRgYbrrA9bOxgad0h8C/7WHgsbgHw/kaDGdgOI+D4aZ9J8uOkzMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMw3LiB/YTK3Y0b+PcnDDw5A8MZGG7cwL84YeDJJb0H+2vbZQkH9hfzEAaGS/c42GPrINMuj4EnZ2C4dAP7GhzEwHAJB/Y4OETSgT39eJmB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMfOsQ7iR4fGNOfvRHm0PgmT+sgdMcgoGnYuAxH83Aw204CANP/rAGTnMIBp6Kgcd8NAMPt+EgDDz5wxo4zSHMM/D93rAbcU4zf9gJx1c2+eOADDzchrcz8D0edrrx1U+rSwq8yWf81efT+3fVXYcbr5/WQz9szIY3qwP/ajjzCtzze8Hmw5fDX7edws3Atz7sQBvejL8Hnw38+unp5E8RA/Nfg98JvM7/X/xle3jO/j7rWt4ofvrtp9Vqvc+f1IufBj/sQBvebrmBt4/7Y+DNw0/7XfayXN0oAh/u2DbuvOJhB9pwCLce9XZNOfozc2rc7Alc3HXIXAV++bjO7v/qc32jCHx4Ej/cU995xcMOtOEszWD0IXvwbvVYB94ddtH8pbe+0XwCr++84mEH2nCWphp92JPI2cDZE2+VcFsl3PYF3hq4ZZrRl+H+3Sso8PYY2D34GvMK3KPagz98ab/KbqvX4G07cH3nxTEFDz50w1maZPT1k3F04N3hAOjt+cOXt+fDjaxe/ntydaN1EJX2b9GDmzbwX/oEv1X5ejjQ/fb8cfC6PJI6/ImHH/In6vcHFTz60A1nafZ7cJydgUuzfw2+Vv4a3H1ns2dMof9GAwc8ynSBD6+/q+Ity0tjCv0XGjjocWb3ZpqBp3ogA4/JwMNtOEsGHm7DWTLwcBvO0uwDH9/oqM71XzgC2pVvesQPKHbDWZpd4NPfqtvXZF0+wA2+AM/AUz1Q6ydnziZVYS/nM3DbzAJ3j4xb12S9fFytHv5Rvuv8+s33q4efsift1eO+eC5f77MtPvwrj5y94/G0b1ysFTqg2A1nKZHA1TVZzdMKr5+yH2VP3tn+vXmsbpdbZCegXj89Hi/WCh5Q7IazNP/AxT3lNVntwE+N5+5vPhfvQ1dbvD1nP8ivBngq36EOHVDshrM0s8DnXoOra7LagdftcrvV6rhF8YPDX4/nEUMHFLvhHL33FvHAzo/hdEStf25fk9UN/HX53Ht4KX748eNp4GqzhQYul/6/U4g+Dm5dk3V2D87//tIJvPA9OKHA+TVZVb6nZrniNTg/w79b9b4Gr5cauH72nH/gdrri2qwiWvb78dvzY7HzHg6LsuInv0WvFx/4bxMY5q3KslZ+bVYZrTwOzi7P+px9oGHTOQ5eLz7wnPfg0SwgcBKvwaMxsIGv3HCOhj/cPev8GCacb8zDJh14OgaGMzCcgeGCA3c33Fy6SueyZR4HTyr8ZMPpz1++/t3lTxoNOaDYDZct9mxSdp7hx+7Z3DEHFLvhsl1xPrj947fnx8P/9s1rbzar8hqcTeNzpO3Lc8o78kt4yvexq9sXBxQ8cjVcc0VH68e76oqb4xcllVfnHD/rnf3Ty8en4ybVHflJw11xYqK6fXFAwSNXQ3TgTX61TnV1TnaasLo6p/E1SvmpweyUcXUmsbyj+pxwFvjkM8MGHlZQ4J53vbL9sNhXm6eF8qtzjt/Gkl9S1zi537ijqFqciWoVNvCwYo+Dt2Xzxpc0VFfn1N+YtKu2qQPXf+jt+fhny9tBDxs8PhWi3qXe579iZX9rnrmvr8452YP3x3O/7W9R2jS+cPj4FXgGHlBZ8n+93l3AKtX2eO1NfXVO62uUcq3LtWrNO44/MfCA4gOXl2RlL8XNPTi/Oqf9W3S9o+bP4uUd+X8f5YVZ1e1qTMGDj531ctRPxlcHPl72vKk/snK8Ouf4jUnZK3Xr97Dyjvy6nfLzDtXtalDBo4+Y8cLUgf/e58YFDPjGpP5BDb7hcsXvwe8K/sak/kENvuGCxb8Gvyv0G5P6xzT4hgs2UuDbxjT4hosWexw84ogG31D9DAxnYDgDwxkYzsBwBoYzMNx7h8hjCh7fmJPX/RkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDwxkYzsBwBoYzMJyB4QwMZ2A4A8MZGM7AcAaGMzCcgeEMDGdgOAPDGRjOwHAGhjMwnIHhDAxnYDgDw/0f7yYNAhM28OoAAAAASUVORK5CYII=" /><!-- --></p>
<p>The package contains many options for simulating fossils and
evaluating the output in a meaningful way requires understanding the
underlying model parameters. Parsing the output for downstream analysis
also requires becoming familiar with the <code>FossilSim</code>
objects.</p>
</div>
</div>
<div id="fossilsim-input" class="section level2">
<h2>FossilSim input</h2>
<p>The starting point for any data generation using
<code>FossilSim</code> is a phylogenetic tree. <code>FossilSim</code>
relies on the widely used <code>ape</code> package <code>phylo</code>
object format for handling trees. The <code>phylo</code> object stores
information about the relationship among branches in a phylogeny in a
matrix called <code>edge</code> and branch lengths are stored in a
vector called <code>edge.length</code>. The age of the tips and internal
nodes can be reconstructed by combining information from the
<code>edge</code> matrix and the <code>edge.length</code> vector. If the
tree has a root edge (i.e. a branch leading to the root) the length of
this edge is stored as a numeric variable called
<code>root.edge</code>.</p>
<p>There are a huge number of packages and options for simulating trees
that can be used as input in <code>FossilSim</code>, including the
<code>ape</code> and <code>TreeSim</code> R packages. An empirical
phylogeny can also be used as input. The only general requirements are
that trees are fully resolved and scaled to time.</p>
</div>
<div id="fossilsim-output" class="section level2">
<h2>FossilSim output</h2>
<p><code>FossilSim</code> can be used to simulate species taxonomy and
fossil sampling times, which are stored and output as the
<code>taxonomy</code> and <code>fossils</code> objects described in the
<strong>“Simulating taxonomy”</strong> and <strong>“Simulating fossils”
vignettes</strong>. The functions used to simulate fossils can take
either a <code>phylo</code> <em>or</em> <code>taxonomy</code> object as
input. This means, in theory, the user never has to interact with the
<code>taxonomy</code> object when simulating fossil data. However, it
may still be useful to become familiar with the concepts underlying the
<code>taxonomy</code> object.</p>
</div>
<div id="what-next" class="section level2">
<h2>What next?</h2>
<p>Information about the <code>taxonomy</code> object and models for
simulating taxonomy can be found in the vignette “Simulating taxonomy”.
Information about the <code>fossils</code> object and models for
simulating fossils can be found in the vignette “Simulating fossils”.
See the <code>paleotree</code> vignette to see how
<code>FossilSim</code> objects can be converted into
<code>paleotree</code> objects.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
